rdflib/container.py

```diff
@@ -0,0 +1,308 @@
+    '''
+    '''
+    def end(self):#
+        # find end index (1-based) of container
+        container = self.uri
+        i=1
+        while True:
+            elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(i)
+
+            if(container, URIRef(elem_uri), None) in self.graph:
+                i+=1
+            else:
+                return i-1
+            
+    def append(self, item):
+        ''' adding item to the end of the container'''
+        
+
+        end = self.end()
+        elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(end+1)
+        container = self.uri
+        self.graph.add((container, URIRef(elem_uri), item))
+        self._len+=1
+        
+        
+
+    def append_multiple(self, other):
+
+        '''adding multiple elements to the container to the end which are in python list other'''
+
+        end = self.end() #it should return the last index
+        
+        container = self.uri
+        for item in other:
+            
+            end+=1
+            self._len+=1
+            elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(end)
+            self.graph.add((container, URIRef(elem_uri), item))
+
+
+        
+
+    def clear(self):
+        '''removing all elements from the container'''
+        container = self.uri
+        graph = self.graph
+        i=1
+        while True:
+            elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(i)
+            if(container, URIRef(elem_uri), None) in self.graph:
+                graph.remove((container, URIRef(elem_uri), None))
+                i+=1
+            else:
+                break
+        self._len=0
+
+        
+
+class Bag(Container):
+    '''unordered container (no preference order of elements)'''
+    def __init__(self, graph, uri, seq=[]):
+        Container.__init__(self,graph,uri,seq,"Bag")
+
+class Alt(Container):
+    def __init__(self, graph, uri, seq=[]):
+        Container.__init__(self,graph,uri,seq,"Alt")
+
+    '''returns any one item of the Alt Container randomly, this functionality added since semantic meaning of ALt is anyone of the elements of the container'''
+    def anyone(self):
+        if (len(self)==0):
+            raise  NoElementException()
+        else:
+            p=randint(1,len(self))
+            item=self.__getitem__(p)
+            return item
+
+class Seq(Container):
+    def __init__(self, graph, uri, seq=[]):
+        Container.__init__(self,graph,uri,seq,"Seq")
+
+    '''adds the item at position pos, valid values are 1 to len+1, this functionality added since sematic meaning of seq is that elements are arranged in decreasing order of preference'''
+    '''user can add the element at the position according to the preference he wants to give to the newly added element '''
+    def add_at_position(self,pos,item):
+        assert isinstance(pos,int)
+        if pos<=0 or pos>(len(self)+1):
+            raise ValueError("Invalid Position for inserting element in rdf:seq")

```

Comment:
Use rdf:Seq (uppercase 'S') to be exact.

Reply:
@niklasl 
I have corrected this.